### PR TITLE
Enabling some HttpListener tests. Disabling UAP and UAPAOT tests against individual classes of bugs. Fixing test string AOT lookups.

### PR DESCRIPTION
**Pull request by CIPop**
Contributes to #17462.

### CODE REVIEW

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 1]
```diff
@@ -255,7 +242,7 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
         public async Task IsWebSocketRequest_GetProperty_ReturnsExpected(string webSocketString, bool expected)
         {
             // Skip on Windows 7 or UAP platforms. WebSocket support is not present.
```
Comment (Diego-Perez-Botero) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Nit: comment needs updating

File: src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj [hunk 2]
```diff
@@ -10,26 +10,26 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='netstandard-Release|AnyCPU'" />
   <ItemGroup>
     <Compile Include="GetContextHelper.cs" />
+    <Compile Include="HttpListenerFactory.cs" />
```
Comment (Diego-Perez-Botero) [inline src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj]: Nit: Was this moved up in order to alphabetize the list of files? If so, it's not in the right place ;)

Comment (CIPop) [inline src/System.Net.HttpListener/tests/System.Net.HttpListener.Tests.csproj]: You're right. These two are just helpers: I reordered this so it's easier to comment out tests.

**Comment by CIPop** [comment]
@mmitche @MattGal from the run script I see that the issue was most likely timeout:
`2017-07-11 22:10:31,548: ERROR: executor(338): _execute_command: Executor timed out after 1200 seconds and was killed.`

Is there a way I could get the test execution log to see which test was hanging?

**Comment by MattGal** [comment]
What you see in https://mc.dot.net/#/user/CIPop/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/dc8ae41f45ab6f705a5dbe0a983d5b46a6a8f4fe/workItem/System.Net.HttpListener.Tests/wilogs is all that made it to the console for the tests that ran.  

Using the repro tool to fetch this test's bits from 272df28f-7f1a-4f7a-8066-a4f69019cfde will give you an idea whether this is an issue with the test itself or the RS2 VMs it's running on.  Note that our VMs have 2 cores and 7 GB of RAM so may be slower than what you use.

**Comment by CIPop** [comment]
@dotnet-bot help

**Comment by CIPop** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug

**Comment by CIPop** [comment]
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build

**Comment by CIPop** [comment]
@dotnet-bot test UWP NETNative x86 Release Build

**Comment by CIPop** [comment]
UWP Outerloop NETNative x86 Release Build

**Comment by CIPop** [comment]
@dotnet-bot test Outerloop UWP NETNative x86 Release Build

**Comment by CIPop** [comment]
I was able to repro the UAP hang in RS2.

**Comment by CIPop** [comment]
@dotnet-bot Test Outerloop Windows x86 Release Build
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop UWP NETNative x86 Release Build

**Comment by CIPop** [comment]
@dotnet-bot Test Outerloop Windows x86 Release Build
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop UWP NETNative x86 Release Build

**Pull request merged**